File: examples/eval-assertion-scoring-override/override.py
from typing import Any, Dict, Optional, Union


def calculate_score(
    named_scores: Dict[str, float], context: Optional[Dict[str, Any]] = None
) -> Dict[str, Union[bool, float, str]]:
    print("Override scoring function (Python):", named_scores)
    accuracy_score = named_scores.get("accuracy", 0)
    return {
        "pass": accuracy_score >= context.get("threshold", 0.7),
        "score": accuracy_score,
        "reason": f"Accuracy-only score: {accuracy_score}",
    }

File: examples/eval-assertion-scoring-override/override.js
const overrideScoring = (namedScores, context) => {
  console.log('Override scoring function (JavaScript):', namedScores);
  const accuracyScore = namedScores.accuracy || 0;
  const fluencyScore = namedScores.fluency || 0;
  const grammarScore = namedScores.grammar || 0;
  const bananaScore = namedScores.banana || 0;

  const minScore = Math.min(accuracyScore, fluencyScore, grammarScore, bananaScore);
  const threshold = context?.threshold ?? 0.7;
  return {
    pass: minScore >= threshold,
    score: minScore,
    reason:
      `minimum score: ${minScore.toFixed(2)} (threshold: ${threshold})\n` +
      `Individual scores:\n` +
      `- contains banana: ${bananaScore.toFixed(2)}\n` +
      `- accuracy: ${accuracyScore.toFixed(2)}\n` +
      `- fluency: ${fluencyScore.toFixed(2)}\n` +
      `- grammar: ${grammarScore.toFixed(2)}`,
  };
};

module.exports = {
  overrideScoring,
};
